### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by calebchris000**
## Highlights:
** As requested by all requirements, I've set up associations.**

Cheers 🍾 .

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  attr_accessor :label, :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students << student
+    student.classroom = self
```
Comment (christianbiring1) [inline classroom.rb:11]: Great job on implementing the `has-many` / `belongs_to` relationship between student and classroom :+1:

File: student_class.rb [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
   end
 
   def play_hooky
     '¯\(ツ)/¯'
   end
+
+  def classroom=(classroom)
+    @classroom = classroom
+    classroom.students.push(self) unless classroom.students.include?(self)
+  end
```
Comment (christianbiring1) [inline student_class.rb:16]: Great job on implementing the `has-many` / `belongs_to` relationship between student and classroom :+1:

File: book_class.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(date, person)
+    Rental.new(date, person, self)
+  end
```
Comment (christianbiring1) [inline book_class.rb:14]: Great on implementing the `many_to_many` and also `many_through` relationship between `person` and `book` :+1:

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  attr_accessor :date, :person, :book
+
+  def initialize(date, person, book)
+    @date = date
+    @person = person
+    @person.rentals << self
+    @book = book
+    @book.rentals << self
```
Comment (christianbiring1) [inline rental.rb:9]: Great job on implementing the ` belongs_to` relationship between `book`, `person` and `rental` :+1:

File: person_class.rb [hunk 5]
```diff
@@ -24,4 +26,8 @@ def correct_name
   def of_age?
     return false unless @age >= 18
   end
+
+  def add_rental(date, book)
+    Rental.new(date, self, book)
+  end
```
Comment (christianbiring1) [inline person_class.rb:32]: Great job on implementing the `has_many` relationship between  `person` and `rentals`

**Review by christianbiring1 — APPROVED** [review]
Hi @calebchris000 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
# STATUS: APPROVED :tada: 
congraaaa

## Highlights
- [ ] Great job on implementing the associations between the classes through the associations as required in the project's requirements :1st_place_medal: 
- [ ] All linters checks are green :heavy_check_mark: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb (hunk 1 above)
Comment (calebchris000) [reply classroom.rb:11]: Thanks! @christianbiring1

**Pull request merged**